src/fsharp/service/service.fs

```diff
@@ -1217,6 +1227,10 @@ type FSharpChecker(legacyReferenceResolver,
         let userOpName = defaultArg userOpName "Unknown"
         backgroundCompiler.InvalidateConfiguration(options, userOpName)
 
+    member _.UpdateBackgroundDocuments(options: FSharpProjectOptions, docs) =
```

Comment:
Looks like this can't be called concurrently, because underlying IncrementalBuilder setCurrentState